Yosemite/Yosemite/Stores/ProductVariationStore.swift

```diff
@@ -62,6 +75,29 @@ private extension ProductVariationStore {
             }
         }
     }
+
+    /// Updates the product variation.
+    ///
+    func updateProductVariation(productVariation: ProductVariation, onCompletion: @escaping (Result<ProductVariation, ProductUpdateError>) -> Void) {
+        remote.updateProductVariation(productVariation: productVariation) { [weak self] result in
+            switch result {
+            case .failure(let error):
+                onCompletion(.failure(ProductUpdateError(error: error)))
+            case .success(let productVariation):
+                self?.upsertStoredProductVariationsInBackground(readOnlyProductVariations: [productVariation],
+                                                                siteID: productVariation.siteID,
+                                                                productID: productVariation.productID) { [weak self] in
+                                                                    guard let storageProductVariation = self?.storageManager.viewStorage
+                                                                        .loadProductVariation(siteID: productVariation.siteID,
+                                                                                              productVariationID: productVariation.productVariationID) else {
+                                                                                                onCompletion(.failure(.notFoundInStorage))
+                                                                                                return
+                                                                    }
+                                                                    onCompletion(.success(storageProductVariation.toReadOnly()))
```

Comment:
⚠️  I noticed that there's a difference if `self` does not exist anymore:

- If `result` is a failure, then `onCompletion` will be called with the error.
- If `result` is a success, then nothing happens because `upsert*()` will not be called.

I think we should use a `guard let self = self...` in this block to be consistent.

Reply:
ooh good catch on this difference! added in https://github.com/woocommerce/woocommerce-ios/pull/2539/commits/0a372c05c4034812cefeb3ef6395e2d3df225cd5